src/components/c-modal.vue

```diff
@@ -0,0 +1,410 @@
+       */
+      onOutsideClick(): void {
+        if (this.closeOnOutsideClick && this.isOpen) {
+          this.close();
+        }
+      },
+
+      /**
+       * Handler for when the modal open-animation is completed.
+       */
+      onAfterEnter(): void {
+        disableBodyScroll(this.container as HTMLElement, { reserveScrollBarGap: true });
+        this.$emit('open');
+        document.addEventListener('keydown', this.onKeyDown);
+      },
+
+      /**
+       * Handler for when the modal close-animation is completed.
+       */
+      onAfterLeave(): void {
+        this.$el.close();
+        enableBodyScroll(this.container as HTMLElement);
+        document.removeEventListener('keydown', this.onKeyDown);
+      },
+    },
+    // render() {},
+  });
+</script>
+
+<style lang="scss">
+  // @use 'sass:math';
+  @use '../setup/scss/variables';
+  @use '../setup/scss/mixins' as *;
+  // @use '../setup/scss/extends' as *;
+  // @use '../setup/scss/functions' as *;
+
+  .c-modal {
+    $this: &;
+
+    @include z-index(modal);
+
+    &,
+    &::backdrop {
+      position: fixed;
+      top: 0;
+      left: 0;
+      width: 100%;
+      height: 100vh;
+    }
+
+    &::backdrop {
+      @include z-index(back);
+
+      content: '';
+      opacity: 0.75;
+      background-color: variables.$color-grayscale--0;
+    }
+
+    &__container {
+      display: grid;
+      grid-template-columns: 1fr;
+      height: 100%;
+      overflow-y: auto;
+
+      @include media(md) {
+        padding: 10vh 0;
+      }
+    }
+
+    &__inner {
+      display: flex;
+      flex-direction: column;
+      width: 100%;
+      background-color: variables.$color-grayscale--1000;
+      justify-self: center;
+
+      @include media(md) {
+        display: block;
+        align-self: center;
+        max-width: 75vw;
+        transform: translateY(-7.5vh);
+      }
+
+      @include media(lg) {
+        max-width: 66vw;
+      }
+
+      @include media(xl) {
+        max-width: 66vw;
+      }
+    }
+
+    &__header,
+    &__sticky-footer {
+      padding: variables.$spacing--25;
+      border-bottom: 1px solid variables.$color-grayscale--600;
+
+      @include media(md) {
+        padding: variables.$spacing--25 variables.$spacing--50;
+      }
+    }
+
+    &__header {
+      border-bottom: 1px solid variables.$color-grayscale--600;
+    }
+
+    &__header-inner {
+      position: relative;
+    }
+
+    &__content, {
+      flex: 1 0 auto;
+      padding: variables.$spacing--25;
+
+      @include media(md) {
+        padding: variables.$spacing--50;
+      }
+    }
+
+    &__sticky-footer {
+      border-top: 1px solid variables.$color-grayscale--600;
+
+      @include media($down: sm) {
+        position: sticky;
+        bottom: 0;
+        left: 0;
+        width: 100%;
+        background-color: variables.$color-grayscale--1000;
+      }
+    }
+
+    &__title {
+      margin: 0;
+    }
+
+    &__button-close {
+      position: absolute;
```

Comment:
suggestion: can we please use a `flexbox`  setup instead? `absolute` is just very unreliable and requires paddings and margins on other elements to preserve the space for the absolute element.